Sources/NIO/BSDSocketAPI.swift

```diff
@@ -0,0 +1,228 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+#if os(Windows)
+import let WinSDK.IPPROTO_IP
+import let WinSDK.IPPROTO_IPV6
+import let WinSDK.IPPROTO_TCP
+import let WinSDK.SOL_SOCKET
+
+import let WinSDK.IP_ADD_MEMBERSHIP
+import let WinSDK.IP_DROP_MEMBERSHIP
+import let WinSDK.IP_MULTICAST_IF
+import let WinSDK.IP_MULTICAST_LOOP
+import let WinSDK.IP_MULTICAST_TTL
+
+import let WinSDK.IPV6_JOIN_GROUP
+import let WinSDK.IPV6_LEAVE_GROUP
+import let WinSDK.IPV6_MULTICAST_HOPS
+import let WinSDK.IPV6_MULTICAST_IF
+import let WinSDK.IPV6_MULTICAST_LOOP
+import let WinSDK.IPV6_V6ONLY
+
+import let WinSDK.TCP_NODELAY
+
+import let WinSDK.SO_ERROR
+import let WinSDK.SO_KEEPALIVE
+import let WinSDK.SO_LINGER
+import let WinSDK.SO_RCVBUF
+import let WinSDK.SO_RCVTIMEO
+import let WinSDK.SO_REUSEADDR
+import let WinSDK.SO_REUSE_UNICASTPORT
+#elseif os(iOS) || os(macOS) || os(tvOS) || os(watchOS)
+import Darwin
+#else
+import Glibc
+#endif
+
+public enum BSDSocket {
+}
+
+public extension BSDSocket {
+    /// Defines socket option levels.
+    struct OptionLevel: RawRepresentable {
+        public typealias RawValue = CInt
+        public let rawValue: RawValue
+        public init(rawValue: RawValue) {
+            self.rawValue = rawValue
+        }
+    }
+
+    /// Defines configuration option names.
+    struct Option: RawRepresentable {
+        public typealias RawValue = CInt
+        public let rawValue: RawValue
+        public init(rawValue: RawValue) {
+            self.rawValue = rawValue
+        }
+    }
+}
+
+extension BSDSocket.OptionLevel: Equatable {
+}
+
+extension BSDSocket.Option: Equatable {
+}
```

Comment:
Any reason these shouldn't be `Hashable`?

Reply:
Sure, it could conform to `Hashable` as well.  I don't think that the default type'd enumerations conform to `Hashable` which is why I didn't add the conformance, but I've got no problems with that.